library/aes.c

```diff
@@ -71,7 +72,7 @@
 
 #if !defined(MBEDTLS_AES_ALT)
 
-#if defined(MBEDTLS_PADLOCK_C) && defined(MBEDTLS_HAVE_X86) && defined(__GNUC__)
```

Comment:
This looks good but line 570 needs updating too.

We also rely on MBEDTLS_PADLOCK_C in `aesni.[ch]` for `mbedtls_aesni_has_support` - but this is not used by the padlock impl. So I think this can be removed.